services/web-app/src/lib/auth.ts

```diff
@@ -1,17 +1,34 @@
-import { NextAuthOptions } from "next-auth"
-import GithubProvider from "next-auth/providers/github"
+import { DynamoDB, DynamoDBClientConfig } from "@aws-sdk/client-dynamodb";
+import { DynamoDBDocument } from "@aws-sdk/lib-dynamodb"
+import { DynamoDBAdapter } from "@next-auth/dynamodb-adapter"
+import { NextAuthOptions } from "next-auth";
+import GithubProvider from "next-auth/providers/github";
+import { clientId, clientSecret, awsAccessKeyId, awsSecretKey, awsRegion } from "./config";
 
-const clientId = process.env.GITHUB_ID;
-const clientSecret = process.env.GITHUB_SECRET;
+export const dynamoConfig: DynamoDBClientConfig = {
+  credentials: {
+    accessKeyId: awsAccessKeyId as string,
+    secretAccessKey: awsSecretKey as string,
+  },
+  region: awsRegion,
+};
 
-if (!clientId || !clientSecret) {
-  throw new Error('Missing required environment variables GITHUB_ID and/or GITHUB_SECRET.');
-}
-export const authOptions: NextAuthOptions = ({
+const dynamoClient = DynamoDBDocument.from(new DynamoDB(dynamoConfig), {
+  marshallOptions: {
+    convertEmptyValues: true,
+    removeUndefinedValues: true,
+    convertClassInstanceToMap: true,
+  },
+})
+
+export const authOptions: NextAuthOptions = {
   providers: [
     GithubProvider({
-      clientId,
-      clientSecret,
+      clientId: clientId as string,
+      clientSecret: clientSecret as string,
     }),
   ],
-});
\ No newline at end of file
+  adapter: DynamoDBAdapter(dynamoClient, {
+    tableName: "dev-web-app-user-data",
```

Comment:
This should not be hardcoded, you should get the table name from the construct or from sst dynamically, otherwise this will fail as soon as we have a different stack name